src/Pact/Native/Ops.hs

```diff
@@ -569,5 +569,9 @@ shiftDef =  defRNative "shift" go
   <> "i.e. they fill the top bits with 1 if the x is negative and with 0 otherwise."
   )
   where
-    go _ [TLitInteger x,TLitInteger y] = return $ toTerm $ shift x (fromIntegral y)
+    go i [TLiteral (LInteger x) xi,TLiteral (LInteger y) yi]
+      | y > 0 = do
+          z <- powImpl i [TLiteral (LInteger 2) yi, TLiteral (LInteger y) yi]
+          mulImpl i [TLiteral (LInteger x) xi, z]
+      | otherwise = return $ toTerm $ shift x (fromIntegral y)
```

Comment:
The non-positive cases are _still_ implemented using `shift`.  I assume we want to drop legacy `shift` implementation and rely on already existing natives?

Reply:
Negative shift is cheap, because it doesn't allocate memory.